scrapy/exporters.py

```diff
@@ -249,13 +263,20 @@ def _serialize_value(self, value):
             return self.export_item(value)
         if isinstance(value, dict):
             return dict(self._serialize_dict(value))
-        if hasattr(value, '__iter__'):
+        if is_listlike(value):
             return [self._serialize_value(v) for v in value]
-        return self._to_str_if_unicode(value)
+        if self.binary:
+            return to_bytes(value, encoding=self.encoding)
+        else:
+            return to_unicode(value, encoding=self.encoding)
 
     def _serialize_dict(self, value):
         for key, val in six.iteritems(value):
+            key = to_bytes(key) if self.binary else key
```

Comment:
you mean, `_serialize_dict`?

yes, it's called for nested items in `_serialize_value` (function above).

it has an extra call in `_get_serialized_fields` to handle the top-level item keys. ;-)